independent-projects/resteasy-reactive/common/processor/src/main/java/org/jboss/resteasy/reactive/common/processor/JandexUtil.java

```diff
@@ -415,4 +417,42 @@ public static boolean isImplementorOf(IndexView index, ClassInfo info, DotName n
         return isImplementorOf(index, superClass, name, additionalIgnoredSuperClasses);
     }
 
+    public static boolean isAssignableFrom(DotName name1, DotName name2, IndexView index) {
+        // java.lang.Object is assignable from any type
+        if (name1.equals(DOTNAME_OBJECT)) {
+            return true;
+        }
+        // type1 is the same as type2
+        if (name1.equals(name2)) {
+            return true;
+        }
+        // type1 is a superclass
+        return findSupertypes(name2, index).contains(name1);
+    }
+
+    private static Set<DotName> findSupertypes(DotName name, IndexView index) {
+        Set<DotName> result = new HashSet<>();
+
+        Deque<DotName> workQueue = new ArrayDeque<>();
```

Comment:
I have to admit I've never seen this sort of implementation. Usually it's a recursive function. I assume you're flattening it on purpose to save stack space?
Also, this is not efficient because you will iterate through the work queue's inheritance entirely to look for _all_ supertypes instead of stopping as soon as you find it.

Reply:
I pretty much copied this from Arc

Reply:
We can of course optimize, but I didn't want to in the remote case we can at some point merge the various util classes

Reply:
Yeah, I figured this came from _somewhere_ :( I also wondered why it was not using the existing `JandexUtil` from quarkus-core-deployment, but this is not usable from `independent-projects`, same as for Arc. Interestingly, that last util class has `isSubclassOf` but not for interfaces, which means it must already exist in many forms _elsewhere_.

@mkouba any reason why Arc's `isAssignableFrom` uses `findSupertypes` instead of stopping on success?

Reply:
I rewrote this in ArC actually, so I can comment on this. The important thing that ArC does and this utility class does not is memoization. In ArC, we remember the full set of supertypes for each class, which allows us to later answer the "is subtype of" queries instantly - we do the traversal just once.

Reply:
See `io.quarkus.arc.processor.AssignabilityCheck`.

Reply:
Also, the work queue algorithm is very common in compilers, I just mangled the name a long time ago and it kinda stuck. In compilers, it is commonly known as "worklist".